cpp/src/arrow/dataset/scanner.h

```diff
@@ -74,6 +74,9 @@ class ARROW_DS_EXPORT ScanOptions {
   // Projector for reconciling the final RecordBatch to the requested schema.
   RecordBatchProjector projector;
 
+  // Maximum row count for scanned batches.
+  int64_t batch_size = 64 << 10;
```

Comment:
This is a change in behaviour, right? 
As now the batch size depends on your data (eg in case of parquet, each row group results in one record batch), while now this would follow this default?

Reply:
Correct